### ISSUE TITLE
CollectTipSetOfHeightAtLeast new OBO error

### ISSUE THREAD

**Issue by ZenGround0** [filecoin-project/go-filecoin#3027]
### Description
Because of the recent fix in #3026 we are now missing one tipset in the ancestor subchain in the case when the sample height falls on a null block.  This bug should only be an issue in the case a PoSt is submitted at the very last possible block.  In that case validators will incorrectly fail with ["sample height out of range"](https://github.com/filecoin-project/go-filecoin/pull/3026/files#diff-0169262ea0c745894301dc43c1446163R56)

### Acceptance criteria
- Modify `CollectTipSetsOfHeightAtLeast` to handle this OBO error
- Simplify the `GetRecentAncestors` code which has some complexity that will go away when making this change.  Specifically we no longer need two iterators because `CollectTipSetsOfHeightAtLeast` will no longer drop a value needed by `CollectAtMostNTipSets`
- Test this edge case

**Comment by anorth** [comment]
See also #3025

**Comment by ZenGround0** [comment]
@deaswang to clarify a bit to help you with #3250 the major fix that this issue is tracking is to make `CollectTipSetsOfHeightAtLeast` return all tipsets up to and including the first tipset with a height less than or equal to the input height.  The reason is that [this change](https://github.com/filecoin-project/go-filecoin/pull/3026/files#diff-0169262ea0c745894301dc43c1446163R27) in PR #3026 modified the sampling behavior during validation to expect all of these tipsets but did not modify the sampling behavior during tipset collection.

tl;dr `CollectTipSetsOfHeightAtLeast` should traverse up to `minHeight` (as it does now) *and additionally* include the first tipset less than `minHeight` if there is no tipset at `minHeight`.
